kornia/augmentation/container/augment.py

```diff
@@ -197,43 +195,14 @@ def __init__(
             # NOTE: only for images are supported for 3D.
             if isinstance(arg, K.AugmentationBase3D):
                 self.contains_3d_augmentation = True
-        self._transform_matrix: Optional[Tensor] = None
         self.extra_args = extra_args
 
     def clear_state(self) -> None:
-        self._transform_matrix = None
-        self._transform_matrices = []
+        self._reset_transform_matrix_state()
         return super().clear_state()
 
-    @property
-    def transform_matrix(self) -> Optional[Tensor]:
-        # In AugmentationSequential, the parent class is accessed first.
-        # So that it was None in the beginning. We hereby use lazy computation here.
-        if self._transform_matrix is None and len(self._transform_matrices) != 0:
-            self._transform_matrix = self._transform_matrices[0]
-            for mat in self._transform_matrices[1:]:
-                self._update_transform_matrix(mat)
-        return self._transform_matrix
-
-    def _update_transform_matrix_by_module(self, module: Module) -> None:
-        if self._transformation_matrix_arg == "skip":
-            return
-        if isinstance(
-            module, (K.RigidAffineAugmentationBase2D, K.RigidAffineAugmentationBase3D, AugmentationSequential)
-        ):
-            # Passed in pointer, allows lazy transformation matrix computation
-            self._transform_matrices.append(module.transform_matrix)  # type: ignore
-        elif self._transformation_matrix_arg == "rigid":
-            raise RuntimeError(
-                f"Non-rigid module `{module}` is not supported under `rigid` computation mode. "
-                "Please either update the module or change the `transformation_matrix` argument."
-            )
-
-    def _update_transform_matrix(self, transform_matrix: Tensor) -> None:
-        if self._transform_matrix is None:
-            self._transform_matrix = transform_matrix
-        else:
-            self._transform_matrix = transform_matrix @ self._transform_matrix
+    def _update_transform_matrix_for_valid_op(self, module: Module) -> None:
+        self._transform_matrices.append(module.transform_matrix)  # type: ignore
```

Comment:
Under this context, we do not need to check since all the module type check has been done in the augmentation initialization stage. If it is the wrong module, let it crash.